pyproject.toml

```diff
@@ -11,10 +11,80 @@ build-backend = "setuptools.build_meta"
 
 [tool.black]
 line-length = 79
-target-version = ['py38', 'py39']
 skip-string-normalization = true
 
 [tool.isort]
 profile = "black"
 line_length = 79
 multi_line_output = 3
+
+[tool.ruff]
+line-length = 79
+exclude = ["amici_models"]
+extend-include = ["*.ipynb"]
+lint.ignore = [
+    "B027",
+    "D100",
+    "D105",
+    "D107",
+    "D202",
+    "E501",
+    "F403",
+    "F405",
+]
+lint.select = [
+    "F",  # Pyflakes
+    "I",  # isort
+    "D",  # pydocstyle (PEP 257)
+    "S",  # flake8-bandit
+    "B",  # flake8-bugbear
+    "C4",  # flake8-comprehensions
+    "T20",  # flake8-print
+    "W",  # pycodestyle Warnings
+    "E",  # pycodestyle Errors
+    # "ANN",  # flakes-annotations TODO: currently produces ~1500 errors to manual fix
+]
+[tool.ruff.lint.pydocstyle]
+convention = "pep257"
+
+[tool.ruff.lint.per-file-ignores]
+"test/julia/test_pyjulia.py" = ["E402"]
+"pypesto/problem.py" = [
+    "D400",
+    "D205",
+    "D107",
+]
+"pypesto/C.py" = [
+    "D400",
+    "D205",
+    "D107",
+]
+"*/__init__.py" = [
+    "F401",
+    "D400",
+    "D205",
+]
+"pypesto/logging.py" = [
+    "D400",
+    "D205",
+    "D107",
+]
+"test/*" = [
+    "T201",
+    "S101",
+    "D",
+]
+"pypesto/util.py" = [
+    "D400",
+    "D205",
+    "D107",
+]
```

Comment:
Why different docstring rules in different files?

Reply:
* I do not really care about the documentation within `examples`, in many cases I think it is not supposed to be documented the same the code base (hence ignore `D` there)
* Same goes for testing and Jupyter notebooks
* All other files (both */init and the other ones). are the ones that have in their heading the following string which is considered docstring:
```
Header
=====

Short description of either the module or the file
```

This produces exactly D400 and D205, which is why in only those files it is excluded, in all other files it should be checked.

Reply:
Ok